### PR TITLE
Added distribution for AMD and CommonJS

### PR DESCRIPTION
**Pull request by gertjanal**
Run `gulp` to transpile the AMD and CommonJS modules

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+var babel = require('gulp-babel'),
```
Comment (ghost) [inline gulpfile.js]: const

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    browserify = require('browserify'),
+    clean = require('gulp-dest-clean'),
+    source = require('vinyl-source-stream'),
+    buffer = require('vinyl-buffer'),
+    gulp = require('gulp')
+    requirejs = require('gulp-requirejs');
+
+gulp.task('clean-build', function () {
```
Comment (ghost) [inline gulpfile.js]:
```
() => gulp.src('src/*.js', {read: false}).pipe(clean('build'))
```

Comment (ghost) [reply gulpfile.js]: and maybe some more array functions

**Review by ghost — APPROVED** [review]

File: gulpfile.js (hunk 1 above)
Comment (gertjanal) [reply gulpfile.js]: done

File: gulpfile.js (hunk 2 above)
Comment (gertjanal) [reply gulpfile.js]: done

File: gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
+const babel = require('gulp-babel'),
+    browserify = require('browserify'),
+    buffer = require('vinyl-buffer'),
+    clean = require('gulp-dest-clean'),
+    gulp = require('gulp')
```
Comment (nfi-jd) [inline gulpfile.js]:
```suggestion
    gulp = require('gulp'),
```

File: examples/searchStream.html [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <script type="text/javascript" src="./node_modules/lodash/lodash.js"></script>
 <script type="module">
-    import {HanskenClient} from './src/hansken.js';
-   
+    import {HanskenClient} from './src/hansken-js.js';
```
Comment (nfi-jd) [inline examples/searchStream.html:3]: This has to be updated in `README.md` as well

File: gulpfile.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+gulp.task('bundle-amd', gulp.series('clean-dist-amd', 'es6-amd', () => {
+    return requirejs({
+            name: 'hansken-js',
+            baseUrl: 'build/amd',
+            out: 'hansken-js.js'
+        })
+        .pipe(gulp.dest('dist/amd'));
+}));
+
+gulp.task('default', gulp.series('bundle-commonjs', 'bundle-amd'));
```
Comment (nfi-jd) [inline gulpfile.js]: There's quite some code duplication in the `amd` and `commonjs` tasks. And I think using a `:` makes it more clear that a task is running for a specific output type. So I'd suggest something like:

```suggestion
const cleanOutput = (target, modules) =>
    () => gulp.src('src/*.js', {read: false})
        .pipe(clean(`${target}/${modules}`));

const build = (modules) =>
    () => gulp.src('src/**/*.js')
        .pipe(babel({
            presets: [
                ['@babel/preset-env', {modules}]
            ]
        }))
        .pipe(gulp.dest(`build/${modules}`));

['amd', 'commonjs'].forEach(modules => {
    gulp.task(`clean-build:${modules}`, cleanOutput('build', modules));
    gulp.task(`es6:${modules}`, gulp.series(`clean-build:${modules}`, build(modules)));
    gulp.task(`clean-dist:${modules}`, cleanOutput('dist', modules));
});

gulp.task('bundle:commonjs', gulp.series(
    'clean-dist:commonjs',
    'es6:commonjs',
    () => browserify('build/commonjs/hansken-js.js')
        .bundle()
        .pipe(source('hansken-js.js'))
        .pipe(buffer())
        .pipe(gulp.dest('dist/commonjs'))
));

gulp.task('bundle:amd', gulp.series(
    'clean-dist:amd',
    'es6:amd',
    () => requirejs({
        name: 'hansken-js',
        baseUrl: 'build/amd',
        out: 'hansken-js.js'
    }).pipe(gulp.dest('dist/amd'))));

gulp.task('default', gulp.series('bundle:commonjs', 'bundle:amd'));
```

File: examples/searchStream.html (hunk 4 above)
Comment (nfi-jd) [reply examples/searchStream.html:3]: And all the other `.html` files in `examples`

**Review by nfi-jd — CHANGES_REQUESTED** [review]

File: gulpfile.js (hunk 3 above)
Comment (gertjanal) [reply gulpfile.js]: Good find. fixed

File: examples/searchStream.html (hunk 4 above)
Comment (gertjanal) [reply examples/searchStream.html:3]: Hmm, I thought I had updated those already, but you're right.
Fixed

File: gulpfile.js (hunk 5 above)
Comment (gertjanal) [reply gulpfile.js]: It took me some time to understand, but I like it :+1:

**Review by nfi-jd — APPROVED** [review]

**Pull request merged**